Admin/MigrateDLs.ps1

```diff
@@ -0,0 +1,347 @@
+        $OldDG = Get-DistributionGroup $Group
+
+        Write-Host "  Exporting properties of $Group to $LogPath\$Group-SourceProperties.txt"
+        $OldDG | Format-List | Out-File $LogPath\$Group-SourceProperties.txt -Force
+        Add-Content $LogPath\$Group-SourceProperties.txt "###"
+        Add-Content $LogPath\$Group-SourceProperties.txt "Distribution Group Members"
+        Add-Content $LogPath\$Group-SourceProperties.txt "###"
+        Add-Content $LogPath\$Group-SourceProperties.txt "`n"
+
+        (Get-DistributionGroupMember $Group).Name | Out-File $LogPath\$Group-SourceProperties.txt -Append
+
+        Write-Host "  Checking for invalid chars for: $Group"
+        [System.IO.Path]::GetInvalidFileNameChars() | ForEach-Object { $Group = $Group.Replace($_, '_') }
+
+        $OldName = [string]$OldDG.Name
+        $OldDisplayName = [string]$OldDG.DisplayName
+        $OldPrimarySmtpAddress = [string]$OldDG.PrimarySmtpAddress
+        $OldAlias = [string]$OldDG.Alias
+        $OldMembers = (Get-DistributionGroupMember $OldDG.Name).Name
+
+        Write-Host "  Storing existing EmailAddresses on $LogPath\$Group.csv for: $Group"
+        "EmailAddress" > "$LogPath\$Group.csv"
+        $OldDG.EmailAddresses.ProxyAddressString >> "$LogPath\$Group.csv"
+        "x500:" + $OldDG.LegacyExchangeDN >> "$LogPath\$Group.csv"
+
+        Write-Host "  Importing the resultant file ($LogPath\$Group.csv) for $Group"
+        $OldAddresses = @(Import-Csv "$LogPath\$Group.csv")
+        $NewAddresses = $OldAddresses | ForEach-Object { $_.EmailAddress.Replace("X500", "x500") }
+
+        Write-Host "  Checking if the group is already created in EXO"
+        if (((Get-OnlineDistributionGroup $Group -ErrorAction 'SilentlyContinue').IsValid) -eq $true) {
+            Write-Host "  $Group already exists. Jumping to the next group on the CSV file"
+            Stop-Transcript
+            Rename-Item $LogPath\$group-log.txt $LogPath\Failed-$group-log.txt -Force
+            $Failed++
+            Write-Host "  Process FAILED and details can be found at $LogPath\Failed-$group-log.txt" -ForegroundColor Red
+            Write-Host ""
+            continue
+        }
+
+        Write-Host "  Creating Group: $Group"
+        New-OnlineDistributionGroup `
+            -Name "$OldName" `
+            -Alias "$OldAlias" `
+            -DisplayName "$OldDisplayName" `
+            -PrimarySmtpAddress "$OldPrimarySmtpAddress" | Out-Null
+
+        if (((Get-OnlineDistributionGroup $Group -ErrorAction 'SilentlyContinue').IsValid) -eq $true) {
+            Write-Host "  $Group was successfully created on EXO"
+
+            Write-Host "  Setting EmailAddresses for: $Group from the imported CSV file ($LogPath\$Group.csv)"
+            Set-OnlineDistributionGroup -Identity "$Group" -EmailAddresses @{Add = $NewAddresses }
+
+            Write-Host "  Setting mail delivery restriction/allowance and members for: $Group"
+            $ir = 0
```

Comment:
Looks like this variable should be named $errorCount maybe? Please give it a meaningful name.